### PR TITLE
Adds rock-salt slug shells, intended for riot and anti-occult use.

### PR DESCRIPTION
**Pull request by SonixApache**
## What this does
adds salted shotgun slugs
they do some (10) brute, mostly agony (20), inject holywater and holysalt (5u each)
holy water/salt does some extra damage to constructs - juggernauts are immune due to their innate armor against low-brute impacts, but all other constructs take an additional 30 damage from the 5u holy water / holy salt in this projectile
salt melts borers inside of people - 2 slugs kills a borer inside a person
salt does no extra damage to borers, so these slugs do their normal brute damage to borers and no more (borers have 20hp)

i assume this affects vampires and zombies and whatever other mobs are affected by holy water / holy salts as well but I haven't tested any of that

## Why it's good
demons begone

## Changelog
:cl:
 * rscadd: Added rock-salt slug shells, intended for riot and anti-occult use. Ask Centcomm for some.

### CODE REVIEW

File: code/modules/projectiles/projectile/bullets.dm [hunk 1]
```diff
@@ -1094,3 +1094,21 @@
 	..()
 	reagents.add_reagent(DIABEETUSOL, 4)
 	reagents.add_reagent(SUGAR, 5)
+
+/obj/item/projectile/bullet/rocksalt
+	name = "rock-salt slug"
+	icon_state = "rsshell"
+	damage = 10
+	agony = 20
+	penetration = 1
```
Comment (DeityLink) [inline code/modules/projectiles/projectile/bullets.dm:1103]: ????

**Comment by DeityLink** [comment]
I know you put holy salts in there because you want the name to matter but it's really not what that reagent was made for in the first place.

**Comment by DeityLink** [comment]
I support this PR otherwise because it's actually a security nerf if they use that instead of lethal shells 😏

**Comment by Eneocho** [comment]
Requesting that you can grind these shells in a mortar to get holy salts and/or holy water.

**Comment by nervere** [comment]
Cult's big weakness is their inability to defend against or produce their own ranged attacks. Adding a way for security to disable all cult abilities from a range is unfairly imbalanced against them.

**Comment by anonymoose-contributor** [comment]
> Cult's big weakness is their inability to defend against or produce their own ranged attacks. Adding a way for security to disable all cult abilities from a range is unfairly imbalanced against them.

You're acting like cultists can't occupy cargo/sci, silently raid the armory, or craft makeshift ranged weapons. Cult does not need to exist in a state where its only successful if it is only using its relevant gimmicks to attack the station. A majority of the complaints cult receives is directly because of their gimmicks. Finding less than lethal ways to hinder cult that don't require an officer to get within sniffing distance to subdue a cultist is a worthwhile addition simply because, in theory, this should create more of a desire to deconvert instead of just braining a cultist, moreover it's keeping people in the round and helping to give cult the proper push and pull of power that it needs.

**Comment by SECBATON-GRIFFON** [comment]
make it do more pain damage for people with visible external wounds, because it's salt after all

**Comment by SonixApache** [comment]
> Cult's big weakness is their inability to defend against or produce their own ranged attacks. Adding a way for security to disable all cult abilities from a range is unfairly imbalanced against them.

blood daggers, pylon constructs, throwing swords, soulsword self-propelled throws, goblets of blood, juggernaut charge, or just doing what every other antag does and either stealing ranged weapons or making makeshift ranged weapons
sure, getting hit with the salt will disable some of these abilities briefly but i don't think that's unfair in any real way

**Comment by DeityLink** [comment]
Yeah to be fair cult has a good bunch of ways to deal with threats at a distance. Also I had forgotten to put it in the changelog but Soul Blades now replenish blood even without a shade when they're planted in an altar, so they can be used much more reliably as a ranged weapon, albeit hard to carry if you're not wearing cult robes.

I don't care about the holy water it injects much because Incense is already a lot more dangerous, but I like the concept of a gun that mostly deals pain damage.

@nervere To be fair if you're a cultist and you see someone pull out a shotgun you should reflexively teleport talisman away.

**Comment by djkramer123** [comment]
> Cult's big weakness is their inability to defend against or produce their own ranged attacks. Adding a way for security to disable all cult abilities from a range is unfairly imbalanced against them.

lol

**Comment by ErrorsWindows** [comment]
> To be fair if you're a cultist and you see someone pull out a shotgun you should reflexively teleport talisman away.

to be fair if i'm a cultist and i see someone pull out a shotgun i reflexively pull out my shotgun so we can compare boomsticks, like that greentext of the guy and his girlfriend's father comparing pistols (i am a cargo player and i have never used a cult ability once in the three years i have been here, seriously i have never bothered figuring it out, i think the only one i've used so far is a stun paper? but someone else gave the paper to me so i didn't have to make it beforehand)

**Comment by DeityLink** [comment]
>works on constructs too

ok now that's over the line.

I'm not excluding we do some balancing after people have had time to adjust to the new update. But I give it a few weeks so we can observe how rounds go first. And so far last night proved that shotguns are already doing their job as anti-cult weapons.

image

(granted I don't think every cult is gonna try and do the Tear Reality ritual in the middle of a primary hallway without at least building walls or something)

**Comment by SonixApache** [comment]
For the sake of Deity's unfounded worries about the salt slugs somehow being more hazardous to constructs than regular lethal shotgun slugs, i've pulled them from the cargo crate and thus made them admin-only for now.

**Comment by anonymoose-contributor** [comment]
@DeityLink I don't believe Sonix is out of line for making them a bit of a soft counter for Juggernauts. I guarantee most of the people who play on here are capable of predicting where a slow moving projectile is going to go anyway.

**Comment by DeityLink** [comment]
Oh I'm not worried for juggernauts, especially since their blocking ability lets them shrug off the projectiles until they're cracked. Wraiths on the other hand have nothing but speed and abilities to jump around. Once occult muted they're reduced to a crawl and can't use their abilities anymore. This is the high risk-high reward gamble of trying to hit them with a null rod. 

Also it seems it wasn't committed in the end but sonix initially mentioned on Discord the idea of making them deal 30 extra damage to non-jug constructs, which I believe was really overkill.

Regardless I actually want to test this in the Thunderdome with other players so I all for this PR to go through in its current admin-only state, I'll get a better opinion then.

**Comment by SonixApache** [comment]
juggernauts are, as the OP states, immune to these as long as their armor is intact

>Also it seems it wasn't committed in the end but sonix initially mentioned on Discord the idea of making them deal 30 extra damage to non-jug constructs, which I believe was really overkill.

again; that 30 extra damage is the effect of holy water and holy salts, this damage is present in the current form
you can test the effects of this by modding a drinking glass to have 25 throwforce, adding 5u holywater and holysalts, and throwing it at a construct
the salt shells still do less damage to constructs than a normal lethal shell does

in testing, after a successful hit, a wraith was unable to phase-shift for a solid 3 seconds while continuing to move at full speed

**Comment by anonymoose-contributor** [comment]
> Regardless I actually want to test this in the Thunderdome with other players so I all for this PR to go through in its current admin-only state, I'll get a better opinion then.

Testing it in an unrealistic, enclosed space isn't going to get you reliable results to see if something makes sense with gameplay balance, as that's much too restrictive. I'd suggest adding them as Sonix intended and then dialing them back if they're way too broken- that should provide much more credible testing. It might be wise, while Cult4 is new and exciting, to take a step back and let members of the community suggest valid tweaks or new items for appropriately dispatching Cult. 
> again; that 30 extra damage is the effect of holy water and holy salts, this damage is present in the current form
you can test the effects of this by modding a drinking glass to have 25 throwforce, adding 5u holywater and holysalts, and throwing it at a construct
> the salt shells still do less damage to constructs than a normal lethal shell does

> in testing, after a successful hit, a wraith was unable to phase-shift for a solid 3 seconds while continuing to move at full speed

I see no issues with this outcome and frankly do not understand as to why you did a 180 on your stance after a realization in how they affected constructs.

**Pull request merged**